src/dialects/oracle/query.js

```diff
@@ -0,0 +1,672 @@
+// Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved
+
+'use strict';
+
+const AbstractQuery = require('../abstract/query');
+const SequelizeErrors = require('../../errors');
+const parserStore = require('../parserStore')('oracle');
+const _ = require('lodash');
+const Utils = require('../../utils');
+const { logger } = require('../../utils/logger');
+
+const debug = logger.debugContext('sql:oracle');
+
+class Query extends AbstractQuery {
+  constructor(connection, sequelize, options) {
+    super(connection, sequelize, options);
+    this.options = _.extend(
+      {
+        logging: console.log,
+        plain: false,
+        raw: false
+      },
+      options || {}
+    );
+
+    this.checkLoggingOption();
+    this.outFormat = options.outFormat || this.sequelize.connectionManager.lib.OBJECT;
+  }
+
+  getInsertIdField() {
+    return 'id';
+  }
+
+  getExecOptions() {
+    const self = this;
+    const execOpts = { outFormat: self.outFormat, autoCommit: self.autoCommit };
+
+    // We set the oracledb
+    const oracledb = self.sequelize.connectionManager.lib;
+
+    if (this.isSelectQuery() && this.model) {
+      const fInfo = {};
+      const keys = Object.keys(this.model.tableAttributes);
+      keys.forEach( key => {
+        const keyValue = this.model.tableAttributes[key];
+        if ( keyValue.type.key === 'DECIMAL') {
+          fInfo[key] = { type: oracledb.STRING };
+        }
+      });
+
+      if ( fInfo ) {
+        execOpts.fetchInfo = fInfo;
+      }
+    }
+    return execOpts;
+  }
+  _run(connection, sql, parameters) {
+    const self = this;
+    // We set the oracledb
+    const oracledb = self.sequelize.connectionManager.lib;
+    // We remove the / that escapes quotes
+    if (sql.match(/^(SELECT|INSERT|DELETE)/)) {
+      this.sql = sql.replace(/; *$/, '');
+    } else {
+      this.sql = sql;
+    }
+    const complete = this._logQuery(sql, debug, parameters);
+    const outParameters = [];
+    const bindParameters = [];
+    const bindDef = [];
+
+    // When this.options.bindAttributes exists then it is an insertQuery/upsertQuery
+    // So we insert the return bind direction and type
+    if (parameters !== undefined && this.options.outBindAttributes) {
+      outParameters.push(...Object.values(this.options.outBindAttributes));
+      // For upsertQuery we need to push the bindDef for isUpdate
+      if (this.isUpsertQuery()) {
+        outParameters.push({ dir: oracledb.BIND_OUT });
+      }
+    }
+
+    this.bindParameters = outParameters;
+    // construct input binds from parameters for single row insert execute call
+    // ex: [3, 4,...]
+    if (parameters !== undefined && typeof parameters === 'object') {
```

Comment:
```suggestion
    if (_.isPlainObject(parameters)) {
```

`null` would pass the current condition

Reply:
Yes we looked into it - we replaced it with `Array.isArray(parameters) || _.isPlainObject(parameters)`